data/json/mapgen/earth.json

```diff
@@ -1,4 +1,39 @@
 [
+  {
+    "method": "json",
+    "object": {
+      "rows": [
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................",
+        "........................"
+      ],
+      "terrain": { ".": "t_rock" }
+    },
+    "om_terrain": "deep_rock",
+    "type": "mapgen",
+    "weight": 100
+  },
```

Comment:
I guess you could use `rock_border` om_terrain from newly added Bordered scenario which consist of solid rock too.